### ISSUE TITLE
Ability to edit a thing's category, subcategory, attribute

### ISSUE THREAD

**Issue by elliotBraem** [near-everything/app#50]

**Comment by ameer-khan-ashraf** [comment]
can I work on this?

**Comment by elliotBraem** [comment]
Yeah go ahead, lmk if you need any more details, although I think there are some things I need to mention first:

- [ ] I only want users to be able to edit things from the organize, so when you're on organize, click view thing, this brings you to pages/organize/thing/[id].js. This is where users can edit.
- [ ] Multiple users will be able to edit the same "thing" all at the same time. On conflict, the api will be able to determine which edit gets applied, but in order to minimize conflicts, I think each field should be edited individually, not together with the whole page (I hope this makes sense... be able to edit one by one rather than edit them all together)
- [ ] For now, the Edit field can be a creatable select, just as we use in Category component for collect, bonus points if you can implement this functionality sharing the same "Category" component.

**Comment by ameer-khan-ashraf** [comment]
The user should be able to rename the current category or just change the category from one to another?

**Comment by elliotBraem** [comment]
Just change from one category to another.

In collect you can either select or propose a new category/subcategory/attribute/option
When they get proposed, they are created in the table with is_approved = false.
The idea is that we don't know how people wanna categorize things yet, so let them put in whatever, and then eventually when we see trends, we'll create category/subcategory/attribute/options with is_approved = true.

Eventually when we get those approved labels, then on organize, this is what users can update the category/subcategory/attribute/options to. They can change the current category to a category with is_approved = true. This bit isn't super important rn tho, really just need to have the updating setup
